build.gradle

```diff
@@ -0,0 +1,248 @@
+// TODO NOW schema updater script - commit 5
+// TODO NOW sdk-compile all, java, and docs - commit 6 
+// TODO NOW handle the git commit the same way as auth does - commit 7
+// TODO NOW delete build.xml, , and Makefile - commit 8
+// TODO NOW run tests from Eclipse w/o specifying classpath manually & remove sourceSets & claspath - commit 9
+
+repositories {
+	mavenCentral()
+}
+compileJava {
+	options.release = 11
+}
+
+tasks.withType(Test) {
+	/* 
+	 * TODO TEST Figure out why tests fail without this and remove. Might have something to do
+	 * with the stfuLoggers() call in many of the tests, might kill logging for tests that
+	 * require it
+	 * Although it seems to make Mongo start up correctly as well which is odd
+	 */
+	forkEvery = 1
+	/*
+	  * TODO TEST split tests into mongo wrapper tests & all other tests (incl. integration).
+	  * Set up GHA to run the non-mongo tests with a single version of mongo and run the
+	  * mongo tests with matrixed mongo versions. Combine coverage at the end somehow
+	*/
+	systemProperty "test.cfg", "./test.cfg"
+	maxHeapSize = "3G"
+	testLogging {
+		exceptionFormat = 'full'
+		showStandardStreams = true
+	}
+	filter {
+		// gradle thinks that classes annotated with @RunWith are tests
+		excludeTestsMatching "*Tester"
+	}
+}
+
+tasks.withType(JacocoReport) {
+	reports {
+		xml.required = true
+		csv.required = true
+	}
+}
+
+test {
+	finalizedBy jacocoTestReport
+}
+
+task testQuick(type: Test) {
+	// for Gradle 9.0 compatibility
+	testClassesDirs = testing.suites.test.sources.output.classesDirs
+	classpath = testing.suites.test.sources.runtimeClasspath
+
+	filter {
+		excludeTestsMatching "*LongTest"
+	}
+	finalizedBy "jacocoTestQuickReport" // must be a string, see TODOs at head of file
+}
+
+task jacocoTestQuickReport(type: JacocoReport, dependsOn: testQuick) {
+	sourceSets sourceSets.main
+	executionData(testQuick)
+}
+
+// Custom java project layout
+sourceSets {
+	main {
+		java {
+			srcDirs = ["src"]
+			exclude '**/test/**'
+		}
+	}
+	test {
+		java {
+			srcDirs = ["src"]
+			include '**/test/**'
+		}
+		resources {
+			srcDirs = ["src"]
+			include "**/*.properties"
+			include '**/*.spec'
+			include '**/*instance.*'
+			include '**/*.instance*'
+			include '**/*.html'
+			include '**/*.css'
+			include '**/*.gif'
+			include '**/*.js'
+			include '**/*.png'
+			include '**/*.txt'
+			include '**/*.weirdsuffix'
+		}
+	}
+}
+
+javadoc {
+	options {
+		links "https://docs.oracle.com/en/java/javase/11/docs/api/"
+		links "https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-core/2.9.9/"
+		links "https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/2.9.9/"
+	}
+}
+
+war {
+	webXml = file('war/web.xml')
+}
+
+configurations {
+	// can't directly access testImplementation, so extend and access
+	testimpl.extendsFrom testImplementation
+}
+
+def fromURL = { url, name ->
+	File file = new File("$buildDir/download/${name}.jar")
+	file.parentFile.mkdirs()
```

Comment:
does this check for directory existence first?

Reply:
It works regardless of whether the build directory exists or not, so presumably it's a `mkdir -p` situation.